.github/workflows/pr-checks.yml

```diff
@@ -8,39 +8,39 @@ jobs:
     runs-on: ${{ matrix.os }}  # This is defined by the strategy below
     strategy:
       # The strategy allows to automatically test different os and python version combinations
-      fail-fast: true  # The PR checks should fail fast not to waste GitHub minutes
+      fail-fast: false  # The PR checks should fail fast not to waste GitHub minutes
       matrix:
         # For reproducibility, we define ubuntu-20.04 rather than ubuntu-latest. PR-CHECKS should only test default.
         os: [ ubuntu-20.04 ]
         # PR-CHECKS should only test default python distribution.
-        python-version: [ '3.11' ]
+        python-version: [ '3.8', '3.9', '3.10', '3.11', '3.12' ]  # TODO: Undo this. Just checking everything as part of PR.
```

Comment:
I just changed this to test all 5 versions on the PR. Will change back before merging.